garage/logger/tensorboard_output.py

```diff
@@ -0,0 +1,272 @@
+from os.path import abspath, dirname
+import shutil
+
+import google.protobuf.json_format as json_format
+from jsonmerge import merge
+import numpy as np
+from tensorboard import summary as summary_lib
+from tensorboard.backend.event_processing import (plugin_event_multiplexer as
+                                                  event_multiplexer)
+from tensorboard.plugins.custom_scalar import layout_pb2, metadata
+import tensorflow as tf
+
+from garage import config
+from garage.logger.outputs import LogOutput
+from garage.logger.tabular_input import TabularInput
+from garage.misc.console import mkdir_p
+
+
+class TensorBoardOutput(LogOutput):
+    def __init__(self, log_dir=config.LOG_DIR):
+        self._scalars = tf.Summary()
+        self._scope_tensor = {}
+        self._has_recorded_tensor = False
+        self._has_dumped_graph = False
+
+        self._histogram_ds = {}
+        self._histogram_summary_op = []
+        self._histogram_distribute_list = [
+            'normal', 'gamma', 'poisson', 'uniform'
+        ]
+        self._feed = {}
+
+        self._default_step = 0
+        self._writer = None
+        self._writer_dir = None
+        self._layout_writer = None
+        self._layout_writer_dir = None
+
+        self.set_dir(log_dir)
+
+    @property
+    def types_accepted(self):
+        """The types that the logger may pass to this output."""
+        return tf.Tensor, TabularInput, tuple
+
+    def log_output(self, data, **kwargs):
+        if isinstance(data, tf.Tensor):
+            self.record_tensor(data.name, data)
+        elif isinstance(data, TabularInput):
+            for key, value in data.get_table_dict().items():
+                self.record_scalar(key, value)
+        elif isinstance(data, tuple) and kwargs['record'] == 'histogram':
+            self.record_histogram(data[0], data[1])
+
+    def set_dir(self, dir_name):
```

Comment:
Hmm this API is strange. Why make this mutable? We can just force the user to make a new TensorBoardOutput. That makes the log much simpler.

@naeioi @krzentner